### ISSUE TITLE
Cuda 10: Failed to get device properties, error code: 3

### ISSUE THREAD

**Issue by huljar** [tensorflow/tensorflow#26572]
I am using [keras-retinanet](https://github.com/fizyr/keras-retinanet) (0.5.0) to perform object detection on images using its TensorFlow backend. When I `load_model` their pre-trained COCO model, the GPU is initialized as expected:
```
Using TensorFlow backend.
WARNING:tensorflow:From .../virtualenv/lib64/python3.6/site-packages/tensorflow/python/framework/op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
2019-03-11 11:26:25.063507: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:998] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-03-11 11:26:25.064403: I tensorflow/compiler/xla/service/service.cc:150] XLA service 0x55c2e09b7bb0 executing computations on platform CUDA. Devices:
2019-03-11 11:26:25.064432: I tensorflow/compiler/xla/service/service.cc:158]   StreamExecutor device (0): GeForce GTX 1080 Ti, Compute Capability 6.1
2019-03-11 11:26:25.066831: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2993245000 Hz
2019-03-11 11:26:25.067138: I tensorflow/compiler/xla/service/service.cc:150] XLA service 0x55c2e06ce580 executing computations on platform Host. Devices:
2019-03-11 11:26:25.067163: I tensorflow/compiler/xla/service/service.cc:158]   StreamExecutor device (0): <undefined>, <undefined>
2019-03-11 11:26:25.068038: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1433] Found device 0 with properties:
name: GeForce GTX 1080 Ti major: 6 minor: 1 memoryClockRate(GHz): 1.6705
pciBusID: 0000:01:00.0
totalMemory: 10.92GiB freeMemory: 10.77GiB
2019-03-11 11:26:25.068063: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1512] Adding visible gpu devices: 0
2019-03-11 11:26:25.069446: I tensorflow/core/common_runtime/gpu/gpu_device.cc:984] Device interconnect StreamExecutor with strength 1 edge matrix:
2019-03-11 11:26:25.069467: I tensorflow/core/common_runtime/gpu/gpu_device.cc:990]      0
2019-03-11 11:26:25.069477: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1003] 0:   N
2019-03-11 11:26:25.070214: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1115] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 10481 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1080 Ti, pci bus id: 0000:01:00.0, compute capability: 6.1)
.../virtualenv/lib64/python3.6/site-packages/keras/engine/saving.py:292: UserWarning: No training configuration found in save file: the model was *not* compiled. Compile it manually.
  warnings.warn('No training configuration found in save file: '
```

However, when calling `predict_on_batch` on this model afterwards, the GPU cannot be initialized any more:
```
2019-03-11 11:26:52.699096: E tensorflow/core/grappler/clusters/utils.cc:83] Failed to get device properties, error code: 3
2019-03-11 11:26:52.947157: W ./tensorflow/core/grappler/optimizers/graph_optimizer_stage.h:241] Failed to run optimizer ArithmeticOptimizer, stage RemoveStackStridedSliceSameAxis node clipped_boxes/strided_slice_2. Error: Pack node (boxes/stack) axis attribute is out of bounds: 2 
2019-03-11 11:26:52.947225: W ./tensorflow/core/grappler/optimizers/graph_optimizer_stage.h:241] Failed to run optimizer ArithmeticOptimizer, stage RemoveStackStridedSliceSameAxis node clipped_boxes/strided_slice_3. Error: Pack node (boxes/stack) axis attribute is out of bounds: 2 
2019-03-11 11:26:52.947245: W ./tensorflow/core/grappler/optimizers/graph_optimizer_stage.h:241] Failed to run optimizer ArithmeticOptimizer, stage RemoveStackStridedSliceSameAxis node clipped_boxes/strided_slice_4. Error: Pack node (boxes/stack) axis attribute is out of bounds: 2 
2019-03-11 11:26:52.947271: W ./tensorflow/core/grappler/optimizers/graph_optimizer_stage.h:241] Failed to run optimizer ArithmeticOptimizer, stage RemoveStackStridedSliceSameAxis node clipped_boxes/strided_slice_5. Error: Pack node (boxes/stack) axis attribute is out of bounds: 2 
2019-03-11 11:26:53.215964: W ./tensorflow/core/grappler/optimizers/graph_optimizer_stage.h:241] Failed to run optimizer ArithmeticOptimizer, stage RemoveStackStridedSliceSameAxis node clipped_boxes/strided_slice_2. Error: Pack node (boxes/stack) axis attribute is out of bounds: 2 
2019-03-11 11:26:53.216024: W ./tensorflow/core/grappler/optimizers/graph_optimizer_stage.h:241] Failed to run optimizer ArithmeticOptimizer, stage RemoveStackStridedSliceSameAxis node clipped_boxes/strided_slice_3. Error: Pack node (boxes/stack) axis attribute is out of bounds: 2 
2019-03-11 11:26:53.216042: W ./tensorflow/core/grappler/optimizers/graph_optimizer_stage.h:241] Failed to run optimizer ArithmeticOptimizer, stage RemoveStackStridedSliceSameAxis node clipped_boxes/strided_slice_4. Error: Pack node (boxes/stack) axis attribute is out of bounds: 2 
2019-03-11 11:26:53.216068: W ./tensorflow/core/grappler/optimizers/graph_optimizer_stage.h:241] Failed to run optimizer ArithmeticOptimizer, stage RemoveStackStridedSliceSameAxis node clipped_boxes/strided_slice_5. Error: Pack node (boxes/stack) axis attribute is out of bounds: 2 
2019-03-11 11:26:53.354507: E tensorflow/stream_executor/cuda/cuda_driver.cc:1131] failed to enqueue async memcpy from host to device: CUDA_ERROR_NOT_INITIALIZED: initialization error; GPU dst: 0x7fc4f4b62900; host src: 0x7fc77b201a00; size: 4=0x4
```

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): no custom code, but TensorFlow is called through keras-retinanet
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Gentoo (Base System release 2.6)
- TensorFlow installed from (source or binary): binary from PyPi
- TensorFlow version: 1.13.1
- Python version: 3.6.5
- CUDA/cuDNN version: 10.0.130 / 7.4.2.24
- GPU model and memory: Nvidia GeForce GTX 1080 Ti (11 GB memory)
- Nvidia driver version: 418.43

**Describe the current behavior**
see above

**Describe the expected behavior**
There should be no error initializing the GPU when using `predict_on_batch`. On other machines I have used older versions (TF 1.10.0 with CUDA 9.0) and it worked fine.

**Code to reproduce the issue**
* Install CUDA 10, cuDNN 7.4.2
* Install tensorflow-gpu 1.13.1 (or 1.13.0rc1/rc2) and keras-retinanet 0.5.0 into a virtualenv from PyPi
* Download pre-trained COCO model from [here](https://github.com/fizyr/keras-retinanet/releases/download/0.5.0/resnet50_coco_best_v2.1.0.h5)
* Load the model using `model = keras_retinanet.models.load_model('path/to/model.h5')` (will generate first blob of output above)
* Predict bounding boxes on an image via `model.predict_on_batch(np.expand_dims(my_image, axis=0))` (will generate second blob of output above)

**Other info / logs**
I am reporting this here and not at the keras-retinanet project because it seems to be a TensorFlow internal issue. Furthermore, it has worked fine with the same version of keras-retinanet when using older versions of TF and CUDA (see above).

Thank you!

**Comment by v-qjqs** [comment]
I'm experiencing similar issue with object detection model training: Failed to get device properties. Unable to determine the device handle for GPU 0000:01:00.0: GPU is lost.  Reboot the system to recover this GPU
OS: notebook with win10
TensorFlow version: tf-gpu 1.13.1
Python version: 3.7.1
CUDA/cuDNN version: 10.0.130 / 7.5.0
GPU model and memory: Nvidia GeForce GTX 1060 (6 GB memory)

**Comment by shaulimaya** [comment]
I have the same issue while running  keras-retinanet after upgraded to cuda 10 t.f. 1.13.1 python 3.6.8

**Comment by FantasyJXF** [comment]
Same here running my Tensorflow multiprocess code with cuda 10, TF 1.13.1, python 3.6.8

**Comment by bharathts1507** [comment]
I am facing the similar issues when loading model for object detection and using multiprocessing

**Comment by yijunsmile** [comment]
i have the same issue, have you deal with it ?

**Comment by jvishnuvardhan** [comment]
This is a stale issue. If this is still an issue, can you please share a simple standalone code to reproduce the issue? Thanks!
